codegenerator/cli/templates/dynamic/codegen/src/Types.res.hbs

```diff
@@ -228,26 +228,34 @@ module HandlerTypes = {
     handler: handler<'eventArgs, 'loaderReturn>,
     wildcard?: bool,
     eventFilters?: SingleOrMultiple.t<'eventFilter>,
+    preRegister?: bool,
   }
 
   @genType
   type eventConfig<'eventFilter> = {
     wildcard?: bool,
     eventFilters?: SingleOrMultiple.t<'eventFilter>,
+    preRegister?: bool,
```

Comment:
```suggestion
    shouldPreRegisterDynamicContracts?: bool,
```
(and everywhere else)

 I think it is good to have consistent naming on this.  I assume you want it to be less verbose in the config and more verbose in the code.  Maybe then we just use preregister everywhere instead of the longer variable. I don't think it is unclear to just be `preRegister`

Reply:
IMO I prefer the verbose naming since it's quite an esoteric feature.

The shortened version I just used for the user API since I don't think they should have such a verbose flag. In the same way, we use `isWildcard` for our internal config everywhere but for the user API it's simply `wildcard`

I would be open to renaming the the user facing config flag but would prefer to keep the verbose name everywhere else.

Reply:
Cool, maybe middle ground - `preRegisterDynamicContracts` everywhere? 

 I don't really agree that the variable will be confusing, especially since the variable comes from the user's handlers and will be in our box, etc., but I think we can make it more descriptive

Reply:
@JasoonS for the user API we have `Contract.Event.contractRegister` so maybe even `preRegisterContracts` I was thinking?

Still prefer verbose naming with "should" or "is" prefix in internal code to imply boolean type.